### PR TITLE
Grappling hook pickup

### PR DESCRIPTION
**Pull request by Guysus**

### CODE REVIEW

**Review by Saretzky — CHANGES_REQUESTED** [review]
3 questions:
1. Is part of the task to add functionality, or is it just the pickup?
2. Has this been tested?
3. is there a grapple input action and set up in the IMC?

**Comment by Guysus** [comment]
> 3 questions:
> 
>     1. Is part of the task to add functionality, or is it just the pickup?
> 
>     2. Has this been tested?
> 
>     3. is there a grapple input action and set up in the IMC?

I think it's just for the pick up, but I was setting up to functions to call.

**Comment by Saretzky** [comment]
> > 3 questions:
> > ```
> > 1. Is part of the task to add functionality, or is it just the pickup?
> > 
> > 2. Has this been tested?
> > 
> > 3. is there a grapple input action and set up in the IMC?
> > ```
> 
> I think it's just for the pick up, but I was setting up to functions to call.

what about questions 2 and 3

**Comment by Guysus** [comment]
> > > 3 questions:
> > > ```
> > > 1. Is part of the task to add functionality, or is it just the pickup?
> > > 
> > > 2. Has this been tested?
> > > 
> > > 3. is there a grapple input action and set up in the IMC?
> > > ```
> > 
> > 
> > I think it's just for the pick up, but I was setting up to functions to call.
> 
> what about questions 2 and 3

I will deleted the input stuff and put it in the player ability branch, then I will test this pick up

File: Source/SteveSloth/Private/MyPlayer.cpp [hunk 1]
```diff
@@ -130,6 +131,13 @@ void AMyPlayer::RemoveEucalyptus(int eucalyptusAmount)
 	PlayerHUD->EucalyptusCountText(EucalyptusCount);
 }
 
+void AMyPlayer::AddGrapplingHook()
+{
+	bIsGrapplingHookUnlocked = true;
+
+	UE_LOG(LogTemp, Warning, TEXT("Add Grappling Hook"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/MyPlayer.cpp:138]: remove testing stuff

File: Source/SteveSloth/Private/ItemBaseClass.cpp [hunk 2]
```diff
@@ -83,6 +85,12 @@ void AItemBaseClass::OnHitboxOverlapBegin(UPrimitiveComponent* OverlappedComp, A
 			case EItemType::RedBud:
 				Steve->SetCurrentHealth(Steve->GetMaxHealth() * AddHealthPercentage);
 			break;
+
+			case EItemType::GrapplingHook:
+				Steve->AddGrapplingHook();
+				UE_LOG(LogTemp, Warning, TEXT("Overlap Grappling Hook"));
```
Comment (Saretzky) [inline Source/SteveSloth/Private/ItemBaseClass.cpp:91]: remove testing stuff

**Review by Saretzky — CHANGES_REQUESTED** [review]
is it working when tested?

**Comment by Jeff-Moreau** [comment]
> is it working when tested?

Nope is having issues with the mesh showing up in game.. the collision is working well though

**Comment by Saretzky** [comment]
> > is it working when tested?
> 
> Nope is having issues with the mesh showing up in game.. the collision is working well though

I think i solved it, just testing right now

**Review by Saretzky — APPROVED** [review]
looks good

**Review by Jeff-Moreau — APPROVED** [review]
Looks Good thank you elad

**Pull request merged**